ebu_tt_live/ui/user_input_producer/uip.js

```diff
@@ -0,0 +1,570 @@
+         $(document).ready(function(){
+
+            var sequence_numbers = {};
+            var all_sequences = {};
+            var interval_send = null;
+            var scheduled_send_media_clock_offset = null;
+            var interval_running_clock = null;
+
+            if (typeof(Storage) === "undefined") {
+               window.alert("You are using an old browser that does not allow to save data between session. Do not reload this page or you will lose your defined sequences.");
+            }
+
+
+            // Get data from local storage if it exists (we retain some data, like existing sequences and sequence numbers
+            // on page reload. Also works when the browser is closed and reopened.
+            if (localStorage.sequence_selector) {
+               $("#sequence-selector").html(localStorage.sequence_selector);
+            }
+            if (localStorage.all_sequences) {
+               all_sequences = JSON.parse(localStorage.all_sequences);
```

Comment:
@kozmaz87 we could merge it without doing this but then I would ask you to file tidy-up issues to fix it all, and that would probably take longer than just fixing it!

Please could you prioritise 1. fixing up the BDD to be clearer/more reusable, 2. Updating the configurator.rst documentation, 3. the robustness things.

If someone else has time to step in and make the js fixes in the meantime then that could help...